lib/tests/streamlit/legacy_dataframe_styling_test.py

```diff
@@ -14,62 +14,101 @@
 
 """Legacy DataFrame Styler unit tests"""
 
+from typing import List, Optional
+
 import numpy as np
 import pandas as pd
 from parameterized import parameterized
 
 import streamlit as st
-from streamlit.proto.DataFrame_pb2 import CellStyle
+from streamlit.proto.DataFrame_pb2 import CellStyle, DataFrame, Table
 from streamlit.proto.DataFrame_pb2 import CSSStyle
+from streamlit.proto.Element_pb2 import Element
 from tests import testutil
 
 
+def _get_df_proto(element: Element) -> DataFrame:
+    return element.data_frame
+
+
+def _get_table_proto(element: Element) -> Table:
+    return element.table
+
+
 class LegacyDataFrameStylingTest(testutil.DeltaGeneratorTestCase):
     """Tests marshalling of pandas.Styler dataframe styling data
     with both st._legacy_dataframe and st._legacy_table.
     """
 
     @parameterized.expand(
-        [("_legacy_dataframe", "data_frame"), ("_legacy_table", "table")]
+        [(st._legacy_dataframe, _get_df_proto), (st._legacy_table, _get_table_proto)]
     )
-    def test_unstyled_has_no_style(self, element, proto):
-        """A DataFrame with an unmodified Styler should result in a protobuf
-        with no styling data
+    def test_unstyled_has_no_style(self, st_element, get_proto):
+        """A pure DataFrame with no Styler should result in a protobuf
+        with no styling data.
         """
 
-        df = pd.DataFrame({"A": [1, 2, 3, 4, 5]})
+        values = [1, 2, 3, 4, 5]
+        display_values = [None] * 5
+        df = pd.DataFrame({"A": values})
 
-        getattr(st, element)(df.style)
-        proto_df = getattr(self._get_element(), proto)
+        st_element(df)
+        proto_df = get_proto(self._get_element())
+        self._assert_column_display_values(proto_df, 0, display_values)
 
-        rows, cols = df.shape
-        for row in range(rows):
-            for col in range(cols):
-                style = get_cell_style(proto_df, col, row)
-                self.assertEqual(style.display_value, "")
-                self.assertEqual(style.has_display_value, False)
-                self.assertEqual(len(style.css), 0)
+    @parameterized.expand(
+        [(st._legacy_dataframe, _get_df_proto), (st._legacy_table, _get_table_proto)]
+    )
+    def test_default_style_has_style_data(self, st_element, get_proto):
+        """A DataFrame with a default Styler will have styling data."""
+
+        values = [1, 2, 3, 4, 5]
+        display_values = ["1", "2", "3", "4", "5"]
+        df = pd.DataFrame({"A": values})
+
+        st_element(df.style)
+        proto_df = get_proto(self._get_element())
+        self._assert_column_display_values(proto_df, 0, display_values)
```

Comment:
This test asserts the new "we have to always pass display_value along if we have a Styler" change